lib/experimental/blocks.php

```diff
@@ -100,14 +100,17 @@ function gutenberg_render_block_connections( $block_content, $block, $block_inst
 		// Allowlist of blocks that support block connections.
 		// Currently, we only allow the following blocks and attributes:
 		// - Paragraph: content.
+		// - Heading: content.
+		// - Button: text.
 		// - Image: url.
 		$blocks_attributes_allowlist = array(
 			'core/paragraph' => array( 'content' ),
+			'core/heading'   => array( 'content' ),
+			'core/button'    => array( 'text' ),
```

Comment:
Unfortunately, button blocks don't seem to be working correctly. When they have an override they're missing the `<div class="wp-block-button">` wrapper.

It's because the code below does this:
```php
$tag_name     = $tags->get_tag();
$markup       = "<$tag_name>$custom_value</$tag_name>";
$updated_tags = new WP_HTML_Tag_Processor( $markup );
$updated_tags->next_tag();
return $updated_tags->get_updated_html();
```

So it considers the updated html of the block to just be the tag that it's updating and disregards any markup before or after that.

Instead it needs to update the `$original_tags` and `get_updated_html` from that, but I don't think it's supported yet 🤔 

It is mentioned in the block bindings tracking issue - https://github.com/WordPress/gutenberg/issues/54536.
> Support to mutate HTML to change the inner content of an element. Until we have a set_inner_markup function, we can manually mutate the HTML for our experiments.

I guess we need to drop button support until then. 

Image also doesn't work, as the code only updates tag content, not attribute values, and it would also strip the image block `<figure>` wrapper, so I think that should be removed too.

Reply:
@SantosGuillamot opened https://github.com/WordPress/gutenberg/pull/57249 that updates the PHP implementation for block bindings that should fix the handling for the Heading and Button blocks.